### PR TITLE
New: Price dynamic in product list and lines

### PR DESCRIPTION
**Pull request by IonAgorria**
Adds price dynamics usage in products list and when lines are added in proposal/order/invoice

Pull split from #4895

### CODE REVIEW

File: htdocs/comm/propal/class/propal.class.php [hunk 1]
```diff
@@ -649,6 +667,27 @@ function updateline($rowid, $pu, $qty, $remise_percent, $txtva, $txlocaltax1=0.0
...
+			$line->fetch_optionals(); // Fetch extrafields for oldcopy
+
+			$staticline = clone $line;
+
+			$line->oldline = $staticline;
+			$this->line = $line;
+			$this->line->context = $this->context;
+
+			//Fetch product
+			if (!empty($this->line->fk_product))
```
Comment (eldy) [inline htdocs/comm/propal/class/propal.class.php]: What is goal of fetching product here ?
With previous code, we did not do that, and it seems not used with new code too.

Comment (IonAgorria) [reply htdocs/comm/propal/class/propal.class.php]: Commit was common with the resources branch which required product fetching in both add/update, I will remove if necessary

**Comment by BebZ** [comment]
Is the dynamic Price usage well documented in the wiki?
Because this behaviour is really interesting

Hubert ANDRIOLO. FM MEDICAL. 0623101731

Le 13 nov. 2017 10:11 AM, "Ion Agorria" <notifications@github.com> a écrit :

*@IonAgorria* commented on this pull request.
------------------------------

In htdocs/comm/propal/class/propal.class.php
<https://github.com/Dolibarr/dolibarr/pull/7802#discussion_r150479408>:

> @@ -649,6 +667,27 @@ function updateline($rowid, $pu, $qty, $remise_percent, $txtva, $txlocaltax1=0.0
 		if ($this->statut == self::STATUS_DRAFT)
 		{
 			$this->db->begin();
+
+			//Fetch current line from the database and then clone the object
and set it in $oldline property
+			$line = new PropaleLigne($this->db);
+			$line->fetch($rowid);
+			$line->fetch_optionals(); // Fetch extrafields for oldcopy
+
+			$staticline = clone $line;
+
+			$line->oldline = $staticline;
+			$this->line = $line;
+			$this->line->context = $this->context;
+
+			//Fetch product
+			if (!empty($this->line->fk_product))

Commit was common with the resources branch which required product fetching
in both add/update, I will remove if necessary

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Dolibarr/dolibarr/pull/7802#discussion_r150479408>, or mute
the thread

**Comment by eldy** [comment]
This PR is now too old, so i close it. A lot of code may have been validated since the PR and the PR may be obsolete, or too difficult to merge due to other changes. Best choice is too cancel it to avoid to keep PR with too low chances to be validated.

**Pull request closed without merging**